spec/support/download_helper.rb

```diff
@@ -12,7 +12,17 @@ def download
   end
 
   def download_content(format: nil)
-    format == "pdf" ? PDF::Reader.new(download) : File.read(download)
+    case format
+    when "pdf"
+      pdf_file = download
+      pdf_double = Object.new
+      pdf_double.define_singleton_method(:pages) do
+        [OpenStruct.new(text: File.read(pdf_file))]
+      end
+      pdf_double
```

Comment:
Je ne suis pas sûr de comprendre ce qui est fait ici. On ne peut plus utiliser `pdf_reader` ? Même si on stub la réponse de l'API comme tu le fais ? Quel est l'intérêt sinon de définir le pdf de la sorte ?

Reply:
Il y a 2 soucis que j'ai essayé de résoudre :
On ne génére pas vraiment un pdf contrairement à avant ou `wicked_pdf` permettait de le faire. On stub la requête du micro service qui renvoi du texte brut. Sans ca Pdf::Reader renvoi une erreur `MalformedPdf :  'PDF does not contain EOF marker'`
Je défini la méthode .pages car on en avait besoin dans `extract_raw_text` du `PdfHelper`.
J'ai fait beaucoup plus simple. Vu qu'on ne manipule plus vraiment un pdf mais une chaine de caractéres j'ai enlevé `extract_raw_text` et le contenu de `download_content` peut désormais difficilement être plus simple 😄